Feature rule check mode

geof77 opened the pull request:
Check mode support for rules module.

In check mode, the module will only return what modifications it would do, without actually making any modificiation.

Suggestions welcome.

## Pull request type

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?
No check mode

Issue Number: #181

## What is the new behavior?

Allow ansible's --check parameter.

lgetwan commented on plugins/modules/rule.py:
```diff
@@ -388,6 +392,9 @@ def move_rule(module, base_url, headers, rule_id, location):
 
     url = base_url + api_endpoint
 
+    if module.check_mode:
```
When doing a "create rule on top of ruleset", the move_rule() gets a "None" for the rule_id and fails when setting the api_endpoint.
Also, the "get_rule_etag()" fails in that case, because of the invalid rule_id.

geof77 replied:
I reverted my last commit, that should fix these.

lgetwan reviewed (changes requested):
Great idea to enable the check mode! Thanks!
I just found one case (when move_rule() is involved) where it fails in check mode.

geof77 commented:
Just added a message that we are in check mode, like this:
```
TASK [cmkserver : Test rule from collection] ***************************
changed: [ilx00073] => {"changed": true, "id": null, "msg": "Rule created (check_mode: no changes made)"}
```
